src/components/molecules/SectionRenderer/SectionHeader.tsx

```diff
@@ -111,21 +115,45 @@ function SectionHeader(props: SectionHeaderProps) {
             {NamePrefix.Component && (
               <NamePrefix.Component sectionInstance={sectionInstance} onClick={toggleCollapse} />
             )}
-            <S.Name
-              $isSelected={sectionInstance.isSelected && isCollapsed}
-              $isHighlighted={sectionInstance.isSelected && isCollapsed}
-              $isCheckable={Boolean(sectionInstance.checkable)}
-              $nameColor={sectionBlueprint.customization?.nameColor}
-              $nameSize={sectionBlueprint.customization?.nameSize}
-              $nameWeight={sectionBlueprint.customization?.nameWeight}
-              $nameVerticalPadding={sectionBlueprint.customization?.nameVerticalPadding}
-              $nameHorizontalPadding={sectionBlueprint.customization?.nameHorizontalPadding}
-              $level={level}
-              onClick={toggleCollapse}
-            >
-              {name}
-            </S.Name>
-
+            {name === 'K8s Resources' ? (
+              <Popover
+                placement="right"
+                content={<WalkThrough walkThrough={wkContent.stepResource} />}
+                title={<WalkThroughTitle title={wkContent.stepResource.title} />}
+                visible={walkThroughStep === wkContent.stepResource.currentStep}
+                overlayClassName="walkthrough"
+              >
+                <S.Name
+                  $isSelected={sectionInstance.isSelected && isCollapsed}
+                  $isHighlighted={sectionInstance.isSelected && isCollapsed}
+                  $isCheckable={Boolean(sectionInstance.checkable)}
+                  $nameColor={sectionBlueprint.customization?.nameColor}
+                  $nameSize={sectionBlueprint.customization?.nameSize}
+                  $nameWeight={sectionBlueprint.customization?.nameWeight}
+                  $nameVerticalPadding={sectionBlueprint.customization?.nameVerticalPadding}
+                  $nameHorizontalPadding={sectionBlueprint.customization?.nameHorizontalPadding}
+                  $level={level}
+                  onClick={toggleCollapse}
+                >
+                  {name}
+                </S.Name>
+              </Popover>
```

Comment:
I agree with Wito that the above snippets looks cleaner and I would also say that instead of hard-coding the WalkThrough logic/components into the `SectionRenderer` component, we should use the `customization` object from the `SectionBlueprint` to provide a wrapper for the `S.Name`

Reply:
@WitoDelnat , totally agree, thank you for providing your suggestion